test/unit/cookie.test.js

```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
             })).valueOf()).to.eql('this is a cookie value');
         });
     });
+
+    describe('valueOf', function () {
+        it('should handle malformed URI sequence', function () {
+            expect((new Cookie({
+                name: 'blah',
+                value: '%E0%A4%A'
+            })).valueOf()).to.eql('%E0%A4%A');
+        });
+    });
+
+    describe('unparseSingle', function () {
+        var rawCookie = 'GAPS=lol;Path=/;Expires=Sun, 04-Feb-2018 14:18:27 GMT;Secure;HttpOnly;Priority=HIGH';
+        it('should be unparsed properly in case of no cookie name', function () {
+            var unparsedSingle = Cookie.unparseSingle(rawCookie);
+            expect(unparsedSingle).equals('');
+        });
+
+        it('should be unparsed properly in case of valid cookie name', function () {
+            var rawCookie = {
+                    name: 'testCookie',
+                    expires: 1502442248,
+                    hostOnly: false,
+                    httpOnly: false,
+                    key: '_ga',
+                    path: '/',
+                    secure: false,
+                    session: false,
+                    value: 'GA1.2.113558537.1435817423'
+                },
+                unparsedSingle = Cookie.unparseSingle(rawCookie);
+            expect(unparsedSingle).equals(rawCookie.name + '=' + rawCookie.value);
+        });
+    });
+
+    describe('unparse', function () {
+        it('should return empty string if not array of cookies', function () {
```

Comment:
```suggestion
        it('should return an empty string for non-array arguments', function () {
```
and test with multiple types.